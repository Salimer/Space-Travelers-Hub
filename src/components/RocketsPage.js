import { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import styled from 'styled-components';
import fetchRockets from '../redux/rockets/thunks';
import RocketElement from './RocketElement';
import { selectRockets } from '../redux/store';

const RocketsPage = () => {
  const dispatch = useDispatch();

  const { rockets, isLoading, error } = useSelector(selectRockets);

  useEffect(() => {
    if (rockets.length > 0) return;
    dispatch(fetchRockets());
  }, [dispatch, rockets.length]);

  if (error) {
    return (
      <div>
        Something went wrong:
        <br />
        &quot;
        {error}
        &quot;
      </div>
    );
  }

  return (
    <Section>
      {isLoading ? (
        <div>loading...</div>
      ) : (
        <>
          <ul>
            {rockets.map((rocket) => (
              <RocketElement key={rocket.id} rocket={rocket} />
            ))}
          </ul>
        </>
      )}
    </Section>
  );
};

export default RocketsPage;

const Section = styled.section`
ul {
  padding: 0;
}
`;
