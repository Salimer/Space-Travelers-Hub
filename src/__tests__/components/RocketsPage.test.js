import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { configureStore } from '@reduxjs/toolkit';
import RocketsPage from '../../components/RocketsPage';
import rocketsReducer from '../../redux/rockets/rocketsSlice';

const store = configureStore({
  reducer: {
    rockets: rocketsReducer,
  },
});

test('renders RocketsPage correctly', () => {
  const rockets = [
    {
      id: 1,
      rocket_name: 'Falcon 9',
      description: 'A two-stage rocket designed and manufactured by SpaceX.',
      image: 'falcon9.jpg',
      reserved: false,
    },
  ];

  const { container } = render(
    <Provider store={store}>
      <RocketsPage rockets={rockets} />
    </Provider>,
  );

  expect(container.firstChild).toMatchSnapshot();
});
