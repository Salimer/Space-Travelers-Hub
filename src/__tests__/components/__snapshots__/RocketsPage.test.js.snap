// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders RocketsPage correctly 1`] = `
<section
  class="sc-gTRrQi bCZDSB"
>
  <div>
    loading...
  </div>
</section>
`;
